pkg/identity/deploy.go

```diff
@@ -649,26 +650,156 @@ func roleBindingResource(roleBindingName, roleName, serviceAccountName, serviceA
 	}
 }
 
-func ensureCRD(ctx context.Context, clientset apiextensionsclientset.Interface, crd apiextensionsv1beta1.CustomResourceDefinition) error {
-	_, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(ctx, crd.Name, metav1.GetOptions{})
+func ensurePostgresSecret(ctx context.Context, clientset kubernetes.Interface, namespace string) error {
+	secret := postgresSecretResource(DexPostgresSecretName)
+
+	_, err := clientset.CoreV1().Secrets(namespace).Get(ctx, DexPostgresSecretName, metav1.GetOptions{})
 	if err != nil {
 		if !kuberneteserrors.IsNotFound(err) {
-			return errors.Wrapf(err, "failed to get existing crd %s", crd.Spec.Names.Plural)
+			return errors.Wrap(err, "failed to get existing secret")
 		}
 
-		_, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(ctx, &crd, metav1.CreateOptions{})
+		_, err = clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
 		if err != nil {
-			return errors.Wrapf(err, "failed to create crd %s", crd.Spec.Names.Plural)
+			return errors.Wrap(err, "failed to create secret")
 		}
 
 		return nil
 	}
 
-	// no patch necessary
+	// no patch needed
+
+	return nil
+}
+
+func postgresSecretResource(secretName string) *corev1.Secret {
+	generatedPassword := ksuid.New().String()
+
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   secretName,
+			Labels: kotsadmtypes.GetKotsadmLabels(AdditionalLabels),
+		},
+		Data: map[string][]byte{
+			"password": []byte(generatedPassword),
+		},
+	}
+}
+
+func ensurePostgresJob(ctx context.Context, clientset kubernetes.Interface, namespace string, registryOptions *kotsadmtypes.KotsadmOptions) error {
+	job := postgresJobResource(DexPostgresJobName, namespace, registryOptions)
+
+	_, err := clientset.BatchV1().Jobs(namespace).Get(ctx, DexPostgresJobName, metav1.GetOptions{})
+	if err != nil {
+		if !kuberneteserrors.IsNotFound(err) {
+			return errors.Wrap(err, "failed to get existing postgres job")
+		}
+
+		_, err = clientset.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+		if err != nil {
+			return errors.Wrap(err, "failed to create job")
+		}
+
+		return nil
+	}
+
+	// TODO: recreate job?
 
 	return nil
 }
 
+func postgresJobResource(name string, namespace string, registryOptions *kotsadmtypes.KotsadmOptions) *batchv1.Job {
+	image := "postgres:10.7"
+	imagePullSecrets := []corev1.LocalObjectReference{}
+	if registryOptions != nil {
+		if s := kotsadmversion.KotsadmPullSecret(namespace, *registryOptions); s != nil {
```

Comment:
this is not used. i wouldn't pass namespace through all these function call just for this. i think i hardcoded this to "whocares" somewhere else

Reply:
it is used in here https://github.com/replicatedhq/kots/blob/master/pkg/docker/registry/registry.go#L92

Reply:
it is not used in the caller